cw/test_capture.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# Simple script to test all relevant capture modes supported by capture.py.
+# To test capture modes on the CW305 board, type
+#
+# ./test_capture.sh cw305
+#
+# For the CW310, simply type
+#
+# ./test_capture.sh
+#
+
+set -e
+set -o pipefail
+
+error () {
+    echo >&2 "$@"
+    exit 1
+}
+
+# Input arguments.
+# Only one argument is supported to specify the FPGA board.
+if [ $# -lt 1 ]; then
+  BOARD=cw310
+else
+  BOARD=$1
+fi
+if `[ "${BOARD}" != "cw305" ] && [ "${BOARD}" != "cw310" ]`; then
+  error "Board ${BOARD} not supported"
+fi
+
+# Create results folder.
+mkdir -p tmp
+
+# AES
+MODE="aes"
+declare -A aes_test_list
+aes_test_list["aes-random"]=100
+aes_test_list["aes-random-batch"]=1000
+aes_test_list["aes-fvsr-key"]=100
+aes_test_list["aes-fvsr-key-batch"]=1000
+
+for test in ${!aes_test_list[@]}; do
+  echo Testing ${test} on ${BOARD}
+  echo Testing ${test} on ${BOARD} - `date` >> "tmp/${BOARD}_test_capture.log"
+  if [ "${test}" = "aes-random" ]; then
+    ARGS="--force-program-bitstream"
+  else
+    ARGS=""
+  fi
```

Comment:
Tests are not always executed in the order they is listed here. Consequently, argument `--force-program-bitstream` is not always set for the first test that is executed. For this reason, some tests end up running using a wrong bitstream.
I would fix this by moving  `ARGS="--force-program-bitstream"` before the for loop and `ARGS=""` at the end of the loop body.

Reply:
Thanks @vrozic for your suggestion, this is now changed.